src/evict/evict_lru.c

```diff
@@ -2499,18 +2527,40 @@ __wt_page_evict_urgent(WT_SESSION_IMPL *session, WT_REF *ref)
     WT_ASSERT(session, !__wt_ref_is_root(ref));
 
     page = ref->page;
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU) || S2BT(session)->evict_disabled > 0)
+
+    if (S2BT(session)->evict_disabled > 0 || F_ISSET_ATOMIC_16(page, WT_PAGE_IN_URGENT_QUEUE))
         return (false);
 
-    /* Append to the urgent queue if we can. */
     cache = S2C(session)->cache;
+
+    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU) && F_ISSET(cache, WT_CACHE_EVICT_ALL))
```

Comment:
why do we need "&&" in these conditions. Also, why do we need to check for "WT_CACHE_EVICT_ALL" ?

What happens to a page that is being requested to be put in the urgent queue, but is already in the ordinary queue, but the cache is not set to evict all at the moment?

Reply:
You can ignore this question - I was implying ```WT_CACHE_EVICT_ALL``` incorrectly.